mlflow/transformers/__init__.py

```diff
@@ -2001,7 +2038,13 @@ def _predict(self, data):
             conversation_output = self.pipeline(self._conversation)
             return conversation_output.generated_responses[-1]
         else:
-            raw_output = self._validate_model_config_and_return_output(data)
+            # If inference task is defined, return tensors instead of generated text
+            model_config = self.model_config.copy()
+            if self.inference_task:
+                model_config.update({"return_tensors": True})
```

Comment:
Can we do this when saving model? It can be confusing that the model returns different thing as it is configured in the file. I guess it's possible because the inference task is declared when saving model.

Reply:
Maybe the comment is misleading here, I intend to only return tensors internally and decode it as generated text + count the tokens within the `predict` function, so the behavior should be the same.